cuebot/src/main/java/com/imageworks/spcue/service/EmailSupport.java

```diff
@@ -228,96 +193,129 @@ public void sendMessage(SimpleMailMessage message) {
+            context.put("succeededFrames", jts.succeeded);
+            context.put("totalFrames", jts.total);
+            context.put("dependFrames", jts.depend);
+            context.put("deadFrames", jts.dead);
+            context.put("waitingFrames", jts.waiting);
+            context.put("eatenFrames", jts.eaten);
+            context.put("failedFrames", jts.dead + jts.eaten + jts.waiting);
+            context.put("checkpointFrames", jts.checkpoint);
             context.put("maxRSS", String.format(Locale.ROOT, "%.1fGB",
                     exj.highMemoryKb / 1024.0 / 1024.0));
-            context.put("coreTime",  String.format(Locale.ROOT, "%.1f",
+            context.put("coreTime", String.format(Locale.ROOT, "%.1f",
                     exj.coreTime / 3600.0));
 
             Template t = ve.getTemplate("/conf/webapp/html/email_template.html");
 
             List<LayerDetail> layers = jobManager.getLayerDetails(job);
             List<LayerStats> layerStats = new ArrayList<LayerStats>(layers.size());
 
-            for (LayerDetail layer: layers)  {
+            boolean shouldCreateFile = false;
+
+            Map<String, byte[]> map = new HashMap<String, byte[]>();
+            loadImage(map, "opencue_logo.png");
+
+            for (LayerDetail layer : layers) {
                 if (layer.type.equals(LayerType.RENDER)) {
                     LayerStats stats = new LayerStats();
                     stats.setDetail(layer);
                     stats.setExecutionSummary(jobManager.getExecutionSummary(layer));
                     stats.setFrameStateTotals(jobManager.getFrameStateTotals(layer));
                     stats.setThreadStats(jobManager.getThreadStats(layer));
-                    stats.setOutputs(jobManager.getLayerOutputs(layer));
+                    stats.setOutputs(jobManager.getLayerOutputs(layer).stream().sorted().collect(Collectors.toList()));
                     layerStats.add(stats);
+                    if (stats.getOutputs().size() > 3)
+                        shouldCreateFile = true;
+                    if (!layer.services.isEmpty())
+                        loadImage(map, "services/" + layer.services.toArray()[0] + ".png");
                 }
             }
 
+            imageMap = Collections.unmodifiableMap(map);
+
             context.put("layers", layerStats);
 
             StringWriter w = new StringWriter();
             t.merge(context, w);
 
             String subject = "OpenCue Job " + d.getName();
-            if (jts.total != jts.succeeded) {
-                subject = "Failed " + subject;
-            }
-            else {
-                subject = "Succeeded " + subject;
+
+            subject = status + subject;
+
+            BufferedWriter output = null;
+            File file = null;
+            if (shouldCreateFile) {
+                try {
+                    logger.info("\n\n\n\nhouldCreateFile \n\n\n\n");
```

Comment:
Remove logger.info()

houldCreateFile -> You mean ShouldCreateFile?

Reply:
```suggestion
```